File: Java/src/test/java/org/codecop/dependencies/c/CheckoutTest.java
package org.codecop.dependencies.c;

import static org.junit.Assert.assertEquals;

import java.math.BigDecimal;

import org.junit.Test;

public class CheckoutTest {

    @Test
    public void createsReceipt() {
        Checkout checkout = new Checkout() {
            @Override
            protected void store(Receipt receipt) {
            }
        };

        Receipt receipt = checkout.createReceipt(new Money(12));

        assertEquals(new BigDecimal(12), receipt.getAmount().asBigDecimal());
    }
}

File: TypeScript/test/c/checkout.test.ts
import { Checkout } from "../../app/c/checkout";
import { Money } from "../../app/c/money";
import { expect } from "chai";

describe('Checkout', () => {

    it('test 3', async () => {
        const checkout = new Checkout();

        await checkout.createReceipt(new Money(12.00))
            .then(receipt => expect(checkout).not.to.be.null);
    });

});
